app/services/foreman/plugin.rb

```diff
@@ -93,6 +93,17 @@ def installed?(id)
 
     def initialize(id)
       @id = id.to_sym
+      add_loggers
+    end
+
+    def add_loggers
+      return unless (defined?(Foreman::Logging) && SETTINGS[@id])
+
+      if (loggers = SETTINGS[@id][:loggers])
```

Comment:
> For the defaults, I am not able to think what defaults a plugin would need to worry about out of the gate. If the plugin is using things like 'Rails.logger' then they will continue to log to the same place and in the same way as before this patch.

It was the custom loggers I was thinking about, so if a plugin decided to use a custom logger to make some of its log messages more categorised, and assuming they should be logged by default, we should ensure the logger is set up even without an explicit config file.

> If we want to, for the purpose of making logs easier to read, start to enforce per-plugin logs then we would need some default log creation if not specified within the settings file for the plugin. This could be based on the Rails engine name for consistency across the board.

I think that'd be a great idea.

> I think splitting the config and the definition of the loggers themselves makes sense across settings and logging. I would expect then, that the 'environmental' configurations would also live under settings (https://github.com/theforeman/foreman/pull/2376/files#diff-d71db24719150ef52a8afd6ef934f9b4R54)?

Yeah, I guess if the rotation settings aren't there then we minimise what's in the general settings file.  My aim is to keep the commonly changed things (log levels) away from rarely changed things (formatting/rotation etc) so misconfigurations (or rather reusing old config files) are less likely to cause bad or missing logging.